src/Nerdbank.MessagePack.Analyzers/KeyAttributeUseAnalyzer.cs

```diff
@@ -50,6 +52,13 @@ public override void Initialize(AnalysisContext context)
 
 		context.RegisterCompilationStartAction(context =>
 		{
+			TypeShape.Roslyn.KnownSymbols typeShapeSymbols = new(context.Compilation);
+			TypeDataModelGenerator typeDataModelGenerator = new(context.Compilation.Assembly, typeShapeSymbols, context.CancellationToken);
```

Comment:
Is this the right place to do this?

Reply:
Seems about right. One thing to note about `TypeDataModelGenerator` is that it has no knowledge of the `GenerateShape` entrypoints or any of the other customization attributes that typeshape proper uses. These have been added in the source generator by extending that class here:

https://github.com/eiriktsarpalis/typeshape-csharp/blob/0b03b2ac2968e39885ba6dfa217b4672bc357cc1/src/TypeShape.SourceGenerator/Parser/Parser.cs